### DISCUSSION TITLE
suggesting AI what the response should be like

### DISCUSSION THREAD

**Discussion by Xeltosh** [SillyTavern/SillyTavern#3212] [Ideas]
Hello,

I am fairly new to local AI-chatting and enjoy the possibilites one can get. My first contact with AI-chatting was [this site](https://perchance.org/ai-character-chat) and it has a function, i really enjoyed:

Anytime you want you can type "/ai (string)" for an response from the character or "/user (string)" to let the AI response in your stead with your typed in suggestion, which then got generated and incorporated into the chat like every other response you or the character normally does. With this you can lead a response into the direction you want the story to go, when the AI doesn't get your hints or directions at all. It also helps with letting a character answer several times.

It also has a passive narrator, which normally starts the chat at the beginning and can be prompted anytime with "/nar (string)" to explain background stuff and so on.

I find it super helpful and miss it already in SillyTavern. Googling for some hours and a quick sweep through the discussions here didn't really help. 
Either i am blind to find it, or it doesn't exist. I hope someone can help or incorporate this in some way.

**Comment by Cohee1207** [comment]
Slash commands exist here too. Welcome to the mysterious world of STscript.

* Crash course documentation: https://docs.sillytavern.app/usage/st-script/
* Look into the following: `/imp` with a prompt, `/sysgen` with a prompt and `/gen` with a prompt coupled with `/sendas`.

**Reply by Xeltosh** [reply]
I tried, but didn't get it to work. Though i have seen something i want to test later.

**Reply by Xeltosh** [reply]
ok, after some playing around i found a way to do it. 

with "/gen name='charactername' as=char (my suggestion i want the AI to use for the resulting message) | /sendas name='charactername' return=pipe " i get what i want, but is there a way to set this up as a slash-command or something like that? I haven't yet understood how quick-replies work, but maybe that could work for my intention? 

I like to work with IT, but i am not a big fan of programming-languages.....XD

**Reply by Cohee1207** [reply]
Yes you can set this up as a quick reply and execute it with /run command. 
It's not a real programming language though. Think of it as Excel formulae.

**Reply by Xeltosh** [reply]
how do i do that? i have read it but haven't got it really XD

and is the quickreply saved even when i close the bot and the command line?
